pkgs/dartpad_ui/lib/main.dart

```diff
@@ -277,7 +278,8 @@ class _DartPadMainPageState extends State<DartPadMainPage>
             sampleId: widget.builtinSampleId,
             flutterSampleId: widget.flutterSampleId,
             channel: widget.initialChannel,
-            fallbackSnippet: Samples.getDefault(type: 'dart'))
+            fallbackSnippet: web.window.localStorage.getItem('user_input') ??
+                Samples.getDefault(type: 'dart'))
```

Comment:
Do you think it's worth it to convert this fallback to a callback, so we don't unnecessarily retrieve the old snippet?

Reply:
I think ?? is short circuiting, so it should be fine?

Reply:
Oh, if you meant not retrieving the old snippet or user input in case a gist id or otherwise is provided. 

I tried making it a callback, and got some very strange errors on the web side. Each time I restarted, it was that JS can't read some property, but it was a different property each time. Once it didn't even let the flutter tool connect because the bootstrapper got stuck. Not sure what's up with that but I'll leave the PR as-is for now

Reply:
@parlough I Implemented this in the latest set of commits. I think there was something wrong with my Flutter installation last time I was working on this. Just upgraded to 3.27.0 with no issues.